### ISSUE TITLE
leading comma + dbt macro mistakenly produces "expected only leading" error

### ISSUE THREAD

**Issue by samLozier** [sqlfluff/sqlfluff#2688]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

on a line like:

```
, 1 as int_col
, {{ mymacro('input') }} as special_column
, 'a' as str_col
```
I get an error: 
```
L:  2 | P:   5 | L019 | Found trailing comma. Expected only leading.  
```
It's only happening on columns with a macro reference and doesn't appear to correlate to the actual content of the macro. By that I mean, that even a macro that returns something as simple as an integer is generating the error. 

### Expected Behaviour

The error should not be raised since the expected leading comma exists

### Observed Behaviour

there appears to be a parsing error that isn't validating the leading comma. 

### How to reproduce

```
-- testfile.sql
select (
'a'
, {{ testmacro() }}
, 1 )

```
```
--testmacro.sql
{% macro testmacro() %}
1
{% endmacro %}
```

yields 
L:   3 | P:   1 | L019 | Found trailing comma. Expected only leading.

### Dialect

dbt + snowflake

### Version

0.10.1

### Configuration

[sqlfluff]
exclude_rules = L031, L033, L034, L023
select_clause_trailing_comma="forbid"
verbose = 0
nocolor = False
dialect = snowflake
templater = dbt
rules = None
recurse = 0
output_line_length = 120
runaway_limit = 10
ignore_templated_areas = True
encoding = autodetect
# Comma separated list of file extensions to lint.

# NB: This config will only apply in the root folder.
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:templater:dbt]
# project_dir = ./src
profiles_dir = ./src/profiles/
profile = snowflake-db

[sqlfluff:indentation]
indented_joins = True
indented_using_on = True
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True
load_macros_from_path= macros/

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable, default='') %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 100
indent_unit = space
comma_style = leading
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# Some rules have their own specific config.
[sqlfluff:rules:L003]
indent_unit=space

[sqlfluff:rules:L007]  # Keywords
operator_new_lines = after

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = consistent

[sqlfluff:rules:L011]  # Aliasing
aliasing = explicit

[sqlfluff:rules:L012]  # Aliasing
aliasing = explicit

[sqlfluff:rules:L014]  # Unquoted identifiers
unquoted_identifiers_policy = all
extended_capitalisation_policy = upper

[sqlfluff:rules:L016]
ignore_comment_lines = False

[sqlfluff:rules:L026]
force_enable = False

[sqlfluff:rules:L028]
force_enable = False

[sqlfluff:rules:L029]  # Keyword identifiers
unquoted_identifiers_policy = aliases
quoted_identifiers_policy = none

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = upper

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = consistent

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = join

[sqlfluff:rules:L047]  # Consistent syntax to count all rows
prefer_count_1 = False
prefer_count_0 = False

[sqlfluff:rules:L052]  # Semi-colon formatting approach.
semicolon_newline = False
require_final_semicolon = False

[sqlfluff:rules:L054]  # GROUP BY/ORDER BY column references.
group_by_and_order_by_style = consistent

ignore:

src/target/
/target
src/dbt_modules/
/dbt_modules
src/dbt_packages/
/dbt_packages
src/great_expectations/
raw_source_queries/
/great_expectations
dbt_packages/
.venv
macros/

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
